### ISSUE TITLE
Project "versions page" returns Internal Server Error

### ISSUE THREAD

**Issue by chaudum** [rtfd/readthedocs.org#1822]
Hi, our main project [crate](https://readthedocs.org/projects/crate/versions/) returns a 500 error on the version page when logged in.
This seems to happen since we move a tag on Github (although I cannot imagine how this is related).

Also, since then new branches and tags are not showing up in the [version admin](https://readthedocs.org/dashboard/crate/versions/). This is not very nice since we cannot build the documentation of the newest release any more.

Update:
We had a branch with a unicode character in the name. We deleted the branch, but the issue still remains.

**Comment by chaudum** [comment]
duplicate of https://github.com/rtfd/readthedocs.org/issues/1410?

**Comment by agjohnson** [comment]
This might be a regression, i thought we had addressed this.

**Comment by rrahn** [comment]
Hi we have the same problem. When clicking on versions on our project page [seqan](https://readthedocs.org/projects/seqan/versions/) we get an server error with a 🐳 saying _Fail. Check back in a bit!_ But we have this state since a couple of days now. 
 We also cannot build the documentation for newer releases.

**Comment by rrahn** [comment]
#2491 seems to be related to this?

**Comment by adamnovak** [comment]
@rrahn I notice your versions page now displays correctly. What did you do to get ReadTheDocs back into a working state after removing the offending characters from branch names?

**Comment by rrahn** [comment]
@adamnovak  to which offending characters do you refer to? The versions page itself does still not work. But I am not so sure, what I could do with it anyway. I would have hoped to find a sync button or something to explicitly force RDT to fetch new tags from the GH repo. The version tab under the admin page suddenly just showed more release tags but is still short of the two most recents tags. I tried to reactivate the Hook in GH several times but still no progress. Not sure, if I have to explicitly trigger something but from the documentation I thought, it simply updates the versions page if a new tag is pushed to the GH repo?

**Comment by rrahn** [comment]
@adamnovak Hmm, ok you probably meant my personal project site here: https://readthedocs.org/projects/rrahn/versions/. That's my personal page not related to the SeqAn organization. But it seems to work perfectly well there. Can it be, that something is broken, due to this weird **§** branch someone accidentally pushed into our main seqan repo? How can I remove it from the version list, as it is garbage anyway?

**Comment by humitos** [comment]
I see a couple of different things here:

1. when the branches from github are updated in RTD?

I'm not sure if they _should_ be updated automatically when they are pushed to github  via a webhook (https://developer.github.com/v3/activity/events/types/#createevent) or not. Is RTD handling this type of event?

I did a simple test in my own local RTD instance by using a local git repository (outside github) and I was able to update the branches in RTD just by creating different branches and re-building the documentation. I mean, as there is no webhook at all, after creating the new branch I pushed the button "Build Version: latest" from my project page and the branch just created appeared without problem.

(this is what is commented in #2491 also)

1. are unicode names in branches supported?

I created a branch with a unicode name and when going to http://localhost:8000/projects/strange-repo/versions/ :bomb: :fire: 

```
[17:22] humitos: strange-repo> git co -b "øð→æ@|-la"
Switched to a new branch 'øð→æ@|-la'
[17:31] humitos: strange-repo> echo "unicodebrach" > docs/index.rst
[17:31] humitos: strange-repo> git ci -am "unicode"
[øð→æ@|-la e8ec8d3] unicode
 1 file changed, 1 insertion(+), 1 deletion(-)
[17:31] humitos: strange-repo> git branch 
  another1677strange_branch)
  master
  new-branch_1
* øð→æ@|-la
[17:33] humitos: strange-repo>
```

Traceback: http://dpaste.com/3D3NK1S

1. are old branches removed from RTD when they are removed from github?

I did the same test locally by creating three branches, b1, b2 and b3. I was able to see all of them under "Versions" and then after removing b3 from my git repository and re-triggered the build for latest b3 disappeared.

Summarizing, to me is not clear _when_ the branches have to be updated in RTD (by specific webhook without building anything or by triggering the latest build -manually or automatic by webhook). On the other hand, at the moment _unicode_ branches names seems to not be supported.
